src/System.ServiceModel.Tests.Common/src/OuterLoopTests.cs

```diff
@@ -0,0 +1,89 @@
+﻿using Newtonsoft.Json;
+using System.IO;
+using System.Net.Http;
+using System.Text;
+
+namespace System.ServiceModel.Tests.Common
+{
+    public abstract class OuterLoopTests : IDisposable
+    {
+        protected string EndpointAddress
+        {
+            get;
+            private set;
+        }
+
+        private string BaseAddress
+        {
+            // Would like to pull this address either from msbuild properties, env vars, or 
+            // configuration passed into xunit.
+            get { return "http://localhost:44283"; }
+        }
+
+        public OuterLoopTests(string resourceName)
+        {
+            MakeConfigRequest();
+            EndpointAddress = MakeResourceRequest(resourceName);
+        }
+
+        public void Dispose()
+        {
+            // Placeholder for releasing the resource
+        }
+
+        private void MakeConfigRequest()
+        {
+            using (HttpClient httpClient = new HttpClient())
+            {
+                httpClient.BaseAddress = new Uri(BaseAddress);
+                var content = new StringContent(
+                    @"{ resourcesDirectory : ""C:\\dotnet\\wcf\\src\\System.Private.ServiceModel\\tests\\Scenarios\\SelfHostWcfService\\bin\\Debug"" }",
+                    Encoding.UTF8,
+                    "application/json");
+                try
+                {
+                    var response = httpClient.PostAsync("/config/", content).Result;
+                    if (!response.IsSuccessStatusCode)
+                        throw new Exception("Unexpected status code: " + response.StatusCode);
+                }
+                catch (Exception exc)
+                {
+                    throw new Exception("Bridge is not running", exc);
+                }
```

Comment:
If the Bridge becomes a singleton, it can record the first failure and fail each test that asks for a resource -- maybe something like Assert.True(Bridge.IsRunning, "The Bridge is not running");